src/pages/Dashboard.js

```diff
@@ -75,6 +84,45 @@ const Dashboard = () => {
   const isFetchingReferralCredits =
     isVerified && !referralCredits && !fetchReferralCreditsError;
 
+  const defaultArray = [];
+  const headers = [
+    { label: "Deposit ID", key: "depositID" },
+    { label: "Amount", key: "depositAmount" },
+    { label: "Reference", key: "reference" },
+    { label: "Bank ID", key: "bankID" },
+    { label: "Created", key: "depositCreated" },
+    { label: "Transfer ID", key: "transferID" },
+    { label: "Coin", key: "coin" },
+    { label: "Amount", key: "cryptoAmount" },
+    { label: "Address", key: "address" },
+    { label: "Created", key: "transferCreated" },
+    { label: "Rate", key: "rate" },
+    { label: "TX", key: "tx" }
+  ];
+
+  const currentYear = new Date().getFullYear();
+
+  const handleDownload = (event, done) => {
+    setDownloadError({ show: false, message: "" });
+    if (year) {
+      const filterTransactions = userTransactions.filter(
+        (ts) =>
+          new Date(ts.depositCreated).getFullYear().toString() ===
+          year.toString()
+      );
+      if (filterTransactions.length > 0) {
+        for (let fts of filterTransactions) {
+          fts.depositID = prefixID(fts.depositID, "D");
+          fts.transferID = prefixID(fts.transferID, "T");
+        }
+        setTransactionsDowload(filterTransactions);
+        done(true);
+      } else {
+        setDownloadError({ show: true, message: "Transactions not found" });
```

Comment:
Not wrong but possibly just a better error message?
```suggestion
        setDownloadError({ show: true, message: "No transactions found" });
```